python/hsml/model.py

```diff
@@ -133,6 +147,19 @@ def save(
         # Returns
             `Model`: The model metadata object.
         """
+
+        if self._model_schema is None:
+            if (
+                self._feature_view is not None
+                and self._training_dataset_version is not None
+            ):
+                features, labels = self._feature_view.get_training_data(
+                    training_dataset_version=self._training_dataset_version
+                )
+                self._model_schema = ModelSchema(
+                    input_schema=features, output_schema=labels
+                )
```

Comment:
I think that having this in the constructor is enough, no need to retry it here.